CMSgov/managed-care-review
# [Draft] CMS user can input unlock reason and see unlock banner with correct data when unlocked.

## Summary
As a CMS user when I click `Unlock submission` a modal opens where I can input a required reason for unlocking submission. Once a submission is unlocked, I will see an unlocked banner with unlock information.

Having issues passing cypress test 
![Screen Shot 2022-03-11 at 11 15 27 AM](https://user-images.githubusercontent.com/98117700/157905803-93605857-e9f1-468d-b13f-f6560232005b.png)

#### Related issues
https://qmacbis.atlassian.net/browse/OY2-12724

#### Screenshots

#### Test cases covered
- unlockStateSubmissions.test.ts
  - unlocked submission revision returns correct `unlockInfo`
  - returns errors if unlocked reason is undefined

- indexSubmission2.test.ts
  - Fixed `unlockTestDraftSubmission` to include `unlockedReason`

- fetchSubmission2.test.ts
  - Fixed `unlockTestDraftSubmission` to include `unlockedReason`

- ReviewAndSubmitModal.test.tsx
  - Renders element with modal hidden
  - Renders element with modal visible with modal title
  - Renders element with modal visible without title

- SubmissionSummary.test.tsx
  - displays unlock banner with correct data for an unlocked submission
  - displays error when submitting without a unlock reason
  - Fixed other tests to input text in unlock modal unlock reason text input.

- unlockResubmit.spec.ts
  - Add input text in unlock modal unlock reason text input.
  - Add test for visible unlock banner with correct data after submission is unlocked.
  - Add test for hidden unlock banner when resubmitted



## QA guidance
- Create and submit a state submission as a state user.
- As a CMS User click `Unlock submission` and type in reason for unlocking. Then submit.
- As a CMS user you will now see a unlock banner on the top of the page with CMS User email, date of unlock and unlock reason
- As a CMS user unlocking submission without inputing an unlock reason should not allow you to submit on the unlock modal.
